channels/OFF-CHAIN.md

```diff
@@ -0,0 +1,503 @@
+
+- `chain_hash` MUST identify the chain to be used
+- `temporary_channel_id` MUST be unique between the involved peers
+- `lock_period` SHOULD be sufficient time to safely publish transactions to the
+  blockchain to stop a cheater
+- `push_amount` MUST be less or equal to `initiator_amount`
+- `initiator_pubkey` MUST be a valid ed25519 pubkey
+
+*Responder* MUST abort if:
+
+- `chain_hash` is unrecognised
+- `initiator_pubkey` not a valid ed25519 pubkey
+- `temporary_channel_id` is not unique between the peers
+
+*Responder* SHOULD abort if:
+
+- `initiator_pubkey` does not have sufficient balance to cover
+  `initiator_amount`
+
+*Responder* MAY abort if:
+
+- `lock_period` is too small
+- `push_amount` is too small
+- `channel_reserve` is too large or small
+- `responder_amount` is too large
+- `initiator_amount` is too small
+
+
+### `channel_accept`
+
+This message is sent by the `responding` peer. It is used to convey the
+conditions under which they are willing to accept the terms proposed by the
+initiating peer.
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| chain_hash           | 32 |
+ ---------------------- ----
+| temporary_channel_id | 32 |
+ ---------------------- ----
+| minimum_depth        | 4  |
+ ---------------------- ----
+| initiator_amount     | 8  |
+ ---------------------- ----
+| responder_amount     | 8  |
+ ---------------------- ----
+| channel_reserve      | 8  |
+ ---------------------- ----
+| responder_pubkey     | 32 |
+ ---------------------- ----
+```
+
+- `chain_hash`: transaction hash of the chain you want to use
+- `temporary_channel_id`: randomly chosen id unique between the involved peers,
+- `minimum_depth`: number of blocks until an opening transaction should be
+  considered final. The `minimum_depth` is set by the responding peer, since
+  they will typically be the one providing a service.
+- `initiator_amount`: amount the initiator is willing to commit
+- `responder_amount`: amount the initiator wants the responder to commit
+- `channel_reserve`: the minimum amount both peers need to maintain. This makes
+  sure that both have to lose something in case they act maliciously
+- `responder_pubkey`: the account that the initiator wants to use to open the
+  channel
+
+(***TODO***: This could be interactive, i.e. if the responding peer sends
+different amounts, then that might communicate that it wants these instead.)
+
+
+#### Requirements
+
+*Responder*:
+
+- `chain_hash` MUST identify the chain to be used
+- `temporary_channel_id` MUST be unique between the involved peers
+- `lock_period` SHOULD be sufficient time to safely publish transactions to the
+  blockchain to stop a cheater
+- `responder_pubkey` MUST be a valid ed25519 pubkey
+
+
+### `funding_created`
+
+In order to open a channel on chain, peers need to cooperate and co-sign the
+`channel_create` transaction. The `funding_created` message is used by the
+initiator to send their signature to the responder.
```

Comment:
In order to restrict the responder ability to arbitrarily choose whether and when to publish his/her signature of the channel_create transaction, shall not the ttl in the channel_create be absolute (i.e. an absolute block height)? (the [description of ttl in channel_create](https://github.com/aeternity/protocol/blob/ffaf834c34d01a6ee4f328de78a9bd0b35153bbe/channels/ON-CHAIN.md#channel_create) suggests ttl as greater than zero that seems relative since the time/height of inclusion of tx in block.)